src/backend/access/transam/xlogrecovery.c

```diff
@@ -4857,7 +4861,7 @@ RecoveryRequiresIntParameter(const char *param_name, int currValue, int minValue
 			ConditionVariableCancelSleep();
 		}
 
-		ereport(FATAL,
+		ereport(recoveryPauseOnMisconfig ? FATAL : WARNING,
 				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
 				 errmsg("recovery aborted because of insufficient parameter settings"),
```

Comment:
This is misleading. You print a WARNING that says "recovery aborted", but it is not aborted.

The GUC name is misleading too: if it's 'on', you abort rather than pause here.

Reply:
Yeh, I tried to minimise changed... Will replace the message.
Can you suggest some better name for the GUC? `allow_replica_misconfig` ?